miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -331,20 +331,30 @@ private NamingScheme getNamingScheme(Sample sample) {
   }
 
   private void updateParentSlides(Sample child, Sample beforeChange, User changeUser) {
-    if (child.getParent() == null || !isSampleSlide(child.getParent()) || !isTissuePieceSample(child)) {
+    if (child.getParent() == null || !isSampleSlide(child.getParent())) {
       return;
     }
     SampleSlide parent = (SampleSlide) deproxify(child.getParent());
-    SampleTissuePiece childPiece = (SampleTissuePiece) child;
-    if (childPiece.getSlidesConsumed() == null) {
+    Integer slidesConsumed = null;
+    if (isTissuePieceSample(child)) {
+      slidesConsumed = ((SampleTissuePiece) child).getSlidesConsumed();
+    } else if (isStockSample(child)) {
+      slidesConsumed = ((SampleStock) child).getSlidesConsumed();
+    }
+    if (slidesConsumed == null) {
       return;
     }
     if (beforeChange == null) {
-      updateParentSlides(parent, parent.getSlides() - childPiece.getSlidesConsumed(), changeUser);
+      updateParentSlides(parent, parent.getSlides() - slidesConsumed, changeUser);
```

Comment:
I tried to test out consuming an absurd number of slides for stock 398048. I expected it to error or update the slides on parent 394809, but that slide parent was unchanged. Am I misunderstanding how the relationships should work?

Reply:
That stock is not parented directly to a slide

Reply:
Thanks!